### PR TITLE
buildtest report --pass

### LINKED ISSUE #1123: [FEATURE]: buildtest report --passed
### Please describe your feature request

Similar to `buildtest report --failure` which returns a list of failed test we can have `buildtest report --passed` which will report all pass tests.

For this test we need to do following
- [x] Add option `buildtest report --passed` in https://github.com/buildtesters/buildtest/blob/devel/buildtest/cli/__init__.py
- [x] Add to bash completion in https://github.com/buildtesters/buildtest/blob/devel/bash_completion.sh
- [x] Implement logic in https://github.com/buildtesters/buildtest/blob/devel/buildtest/cli/report.py
- [x] Add user documentation in https://github.com/buildtesters/buildtest/blob/devel/docs/gettingstarted/query_test_report.rst
- [x] Add regression test in https://github.com/buildtesters/buildtest/blob/devel/tests/cli/test_report.py

### Post question in Slack

- [X] I agree that I posted my question in [slack](https://hpcbuildtest.slack.com/) before creating this issue

### Is there an existing issue

- [X] I confirm there is no existing [issue](https://github.com/buildtesters/buildtest/issues) for this issue

### PR DESCRIPTION
**Pull request by shahzebsiddiqui**

### CODE REVIEW

File: docs/gettingstarted/query_test_report.rst [hunk 1]
```diff
@@ -167,6 +167,21 @@ filter test and report all **FAIL** tests.
 
 .. command-output:: buildtest report --filter tags=tutorials --format name,id,state --failure
 
+Find all Failed Tests (``buildtest report --passed``)
```
Comment (Xiangs18) [inline docs/gettingstarted/query_test_report.rst]: Here should be Find all Pass Tests

Comment (prathmesh4321) [reply docs/gettingstarted/query_test_report.rst]: Okay, fixed it.

File: coverage.json [hunk 2]
```diff
@@ -0,0 +1 @@
...ches": 22, "num_partial_branches": 2, "covered_branches": 20, "missing_branches": 2}, "missing_lines": [30, 62], "excluded_lines": [194, 195, 197, 198]}, "buildtest/utils/timer.py": {"executed_lines": [1, 4, 5, 8, 9, 10, 11, 13, 20, 29], "summary": {"covered_lines": 9, "num_statements": 17, "percent_covered": 52.0, "percent_covered_display": "52.00", "missing_lines": 8, "excluded_lines": 0, "num_branches": 8, "num_partial_branches": 0, "covered_branches": 4, "missing_branches": 4}, "missing_lines": [15, 16, 18, 22, 23, 25, 26, 30], "excluded_lines": []}, "buildtest/utils/tools.py": {"executed_lines": [1, 4, 5, 12], "summary": {"covered_lines": 3, "num_statements": 3, "percent_covered": 100.0, "percent_covered_display": "100.00", "missing_lines": 0, "excluded_lines": 1, "num_branches": 2, "num_partial_branches": 0, "covered_branches": 2, "missing_branches": 0}, "missing_lines": [], "excluded_lines": [12]}}, "totals": {"covered_lines": 3941, "num_statements": 5448, "percent_covered": 70.30327428878154, "percent_covered_display": "70.30", "missing_lines": 1507, "excluded_lines": 174, "num_branches": 2004, "num_partial_branches": 214, "covered_branches": 1298, "missing_branches": 706}}
```
Comment (shahzebsiddiqui) [inline coverage.json]: please remove this file

**Comment by shahzebsiddiqui** [comment]
@prathmesh4321 

Please fix the merge conflict

Screen Shot 2022-06-30 at 8 19 02 PM

**Comment by shahzebsiddiqui** [comment]
@prathmesh4321 please add an entry for this option in `buildtest help report` this is in https://github.com/buildtesters/buildtest/blob/devel/buildtest/cli/help.py

File: coverage.json (hunk 2 above)
Comment (prathmesh4321) [reply coverage.json]: Removed the file.

**Comment by prathmesh4321** [comment]
> @prathmesh4321
> 
> Please fix the merge conflict
> 
> Screen Shot 2022-06-30 at 8 19 02 PM

Done.

**Comment by prathmesh4321** [comment]
> @prathmesh4321 please add an entry for this option in `buildtest help report` this is in https://github.com/buildtesters/buildtest/blob/devel/buildtest/cli/help.py

@shahzebsiddiqui Done!

**Comment by shahzebsiddiqui** [comment]
Thanks @prathmesh4321 this looks good. Great job on your first contribution. I will merge this in.

**Comment by shahzebsiddiqui** [comment]
@prathmesh4321 you may want to check your git configuration the commit history is showing `Prathmesh` and your github username which will lead two different authors in commit history. I know it's you but it can be confusing 

This is fairly easy to setup. You need to use `git config --global user.name` and `git config --global user.email`. See https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup 

Screen Shot 2022-07-01 at 8 41 28 AM

**Pull request merged**